lib/formats/dotnet-core-parser.js

```diff
@@ -1,82 +1,153 @@
+'use strict';
 var Dependency = require('../dependency');
 var debug = require('debug')('snyk');
 
+//TODO: any convention for global vars? (gFreqDeps)
+let freqDeps = [];
+
+function initFreqDepsDict() {
+  freqDeps['Microsoft.NETCore.Platforms'] = false;
+  freqDeps['Microsoft.NETCore.Targets'] = false;
+  freqDeps['System.Runtime'] = false;
+  freqDeps['System.IO'] = false;
+  freqDeps['System.Text.Encoding'] = false;
+  freqDeps['System.Threading.Tasks'] = false;
+  freqDeps['System.Reflection'] = false;
+  freqDeps['System.Globalization'] = false;
+  freqDeps.dependencies = {};  
+}
+
 function convertFromPathSyntax(path) {
-  var name = path.split('/').join('@'); // posix
+  let name = path.split('/').join('@'); // posix
   name = name.split('\\').join('@'); // windows
   return name;
 }
 
 function collectFlatList(targetObj) {
-  var names = Object.keys(targetObj);
-  var libs = names.map(function (name) {
+  const names = Object.keys(targetObj);
+  return names.map(function (name) {
     name = convertFromPathSyntax(name);
     return name;
   });
-  return libs;
 }
 
-function registerDependency(packageTree, resolvedName) {
-  var name = resolvedName.split('@')[0];
-  var version = resolvedName.split('@')[1];
-  packageTree.dependencies[name] =
-    packageTree.dependencies[name] || new Dependency(name, version);
-  debug(resolvedName);
+function buildTreeRecursive(targetDeps, depName, parent, treeDepth) {
+  if (treeDepth > 20)
+  {
+    throw new Error('The depth of the tree is too big.');
+  }
+  
+  let depResolvedName = "";
+  let originalDepKey = "";
+  
+  debug(`${treeDepth}: Looking for '${depName}'`);
+  const exists = Object.keys(targetDeps).some(function (currentDep) {
+      let currentResolvedName = convertFromPathSyntax(currentDep);
+      if(currentResolvedName.split('@')[0].toLowerCase() === depName){
+        depResolvedName = currentResolvedName;
+        originalDepKey = currentDep;
+        debug(`${treeDepth}: Found '${currentDep}'`);
+        return true;
+      }
+  });
+
+  if (!exists) {
+    debug(`Failed to find '${depName}'`);
+    return;
+  }
+
+  const depVersion = depResolvedName.split('@')[1];
+
+  parent.dependencies[depName] = parent.dependencies[depName] || new Dependency(depName, depVersion);
+
+  Object.keys(targetDeps[originalDepKey].dependencies || {}).forEach(function (currentDep)
+  {
+    if (currentDep in freqDeps)
+    {
+      if (freqDeps[currentDep] != false)
+      {
+        return;
+      }
+
+      buildTreeRecursive(targetDeps, currentDep.toLowerCase(), freqDeps, 0);
+      freqDeps[currentDep] = true;
+    }
+    else
+    {
+      buildTreeRecursive(targetDeps, currentDep.toLowerCase(), parent.dependencies[depName], treeDepth + 1);
+    }
+  });
+}
+
+
+function getFrameworkObjToRun(manifest) {
+  if (!manifest.project.frameworks) {
```

Comment:
This can explode if `manifest.project` does not exist. Consider `!_.get(manifest, 'project.frameworks')`

Reply:
I'm checking before 'manifest.project'.
But this is a good practice...:)

Reply:
looks like there is no need to pass in manifest, so pass in `project` or `project.frameworks`